src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -560,36 +560,40 @@ private EnvironmentOptions GetEnvironmentOptions(bool requiresNetFramework, stri
             var os = DetermineOSPlatform();
             EnvironmentOptions options = new EnvironmentOptions();
 
-            if (os == OSPlatform.Linux || os == OSPlatform.OSX)
-            {
-                if (requiresNetFramework)
-                {
-                    options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, Constants.MsBuildCommandName);
-                }
-            }
-
             //We want to provide the MsBuild path only if it's a framework solution. Buildalyzer automatically builds core solutions using "dotnet"
             if (requiresNetFramework)
             {
-                try
+                if (os == OSPlatform.Linux || os == OSPlatform.OSX)
```

Comment:
Why did you move the `if (os == OSPlatform.Linux || os == OSPlatform.OSX)` inside the `if (requiresNetFramework)`?

```
 //We want to provide the MsBuild path only if it's a framework solution. Buildalyzer automatically builds core solutions using "dotnet"
if (requiresNetFramework)
{
    if (os == OSPlatform.Linux || os == OSPlatform.OSX)
```

Is it a valid use case that the OS would be linux if the solution is NetFramework as Framework only runs on Windows?

Reply:
@ppittle @jonlouie 
I was following the old logic
```
            if (os == OSPlatform.Linux || os == OSPlatform.OSX)
            {
                if (requiresNetFramework)
                {
                    options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, Constants.MsBuildCommandName);
                }
            }
```
If `(os == OSPlatform.Linux || os == OSPlatform.OSX)` and `requiresNetFramework` are mutually exclusive, I doubt we ever hit this point.

Reply:
I think we have to be careful about this case. It's true that unix platforms won't build .NET Framework projects, but nothing is stopping developers from trying to analyze a .NET Framework project from a linux/mac machine. Also, look at like 296 of [this diff](https://github.com/aws/codelyzer/commit/97272f24821075a57f82505d14e698c71d690ec4#diff-3ea344cda7ecb78aa25b568af9c5c7024c50d0c6f6ad859dc2f4b779a7bb73b9). The `requiresNetFramework` check was added here intentionally. Unless we fully understand the consequences of removing it, we should let it remain.

To make the intent of this code more clear, I would suggest refactoring the code into a separate function similar to this:
```csharp
if (requiresNetFramework)
{
    SetMSBuildPathEnvironmentVariable(options);
}
.
.
.
private void SetMSBuildPathEnvironmentVariable(EnvironmentOptions options)
{
    var os = DetermineOSPlatform();
    if (os == OSPlatform.Linux || os == OSPlatform.OSX)
    {
        options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, Constants.MsBuildCommandName);
    }
    else  // os == OSPlatform.Windows
    {
        var msbuildExe = _analyzerConfiguration.BuildSettings.MSBuildPath;
        if (string.IsNullOrEmpty(msbuildExe))
        {
            msbuildExe = _msBuildDetector.GetFirstMatchingMsBuildFromPath(toolsVersion: toolsVersion);
        }
        if (!string.IsNullOrEmpty(msbuildExe)) 
        {
            options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, msbuildExe);
        }
        else 
        {
            Logger.LogError("Codelyzer wasn't able to retrieve the MSBuild path. " +
                "Visual Studio and MSBuild might not be installed.");
        } 
    }
}
```

Reply:
I agree with not removing the `requiresNetFramework` check. To your second point - I didn't use this pattern often, in this case it tries to set for `EnvironmentOptions` depending on an outsider parameter `DetermineOSPlatform `. But I see this is how we avoid using `try/catch`. Is there a practice that we should follow regarding the codebase?